tests/framework/extensions/nodes/node_status.go

```diff
@@ -51,61 +55,61 @@ func IsNodeReady(client *rancher.Client, ClusterID string) error {
 	return err
 }
 
-func IsRKE1EtcdNodeReplaced(client *rancher.Client, etcdNodeToDelete management.Node, clusterResp *management.Cluster, numOfEtcdNodesBeforeDeletion int) (bool, error) {
-	numOfEtcdNodesAfterDeletion := 0
-
-	err := wait.Poll(PollInterval, PollTimeout, func() (done bool, err error) {
-		machines, err := client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
-			"clusterId": clusterResp.ID,
+// IsEveryMachineReady is a helper method that will loop and check if the machine object of every node in a non-rke1 cluster is ready.
+// It will return an error if the machine object is not ready after set amount of time.
+func IsEveryMachineReady(client *rancher.Client, clusterID string) error {
```

Comment:
Also, I am probably confused because of godoc the comment; wouldn't this work for rke2/k3s instead `non-rke1`? Would this function work against hosted clusters?